ngx-tools/src/cookie-service/cookie.service.ts

```diff
@@ -0,0 +1,210 @@
+  providedIn: 'root',
+})
+export class TsCookieService {
+  private readonly documentIsAccessible: boolean;
+  private document: Document;
+
+  constructor(
+    // HACK: This `any` is required. See comment inside constructor.
+    @Inject(DOCUMENT) private _document: any,
+    @Inject(PLATFORM_ID) private platformId: InjectionToken<Object>,
+  ) {
+    // HACK: The angular compiler doesn't understand the type `Document` when determining the metadata for injectable parameters.
+    // So we are using `any` for the injection signature (making Angular happy), but coercing to a TypeScript type within the class.
+    // See https://github.com/angular/angular/issues/20351
+    this.document = _document as Document;
+    // The PLATFORM_ID allows us to check if we're in a browser
+    this.documentIsAccessible = isPlatformBrowser(platformId);
+  }
+
+
+  /**
+   * Set a cookie
+   *
+   * @param name - Cookie name
+   * @param value - Cookie value
+   * @param expires - Number of days until the cookies expires or an actual `Date`
+   * @param path - Cookie path
+   * @param domain - Cookie domain
+   * @param secure - Secure flag
+   * @param sameSite - OWASP samesite token `Lax` or `Strict`
+   */
+  public set(
+    name: string,
+    value: string,
+    expires?: number | Date,
+    path?: string,
+    domain?: string,
+    secure?: boolean,
+    sameSite?: 'Lax' | 'Strict',
+  ): void {
+    if (!this.documentIsAccessible) {
+      return;
+    }
+
+    let cookieString: string = encodeURIComponent(name) + '=' + encodeURIComponent(value) + ';';
+
+    if (expires) {
+      if (typeof expires === 'number') {
+        const dateExpires: Date = new Date(new Date().getTime() + expires * 1000 * 60 * 60 * 24);
+
+        cookieString += `expires=${dateExpires.toUTCString()};`;
+      } else {
+        cookieString += `expires=${expires.toUTCString()};`;
+      }
+    }
+
+    if (path) {
+      cookieString += `path=${path};`;
+    }
+
+    if (domain) {
+      cookieString += `domain=${domain};`;
+    }
+
+    if (secure) {
+      cookieString += 'secure;';
+    }
+
+    if (sameSite) {
+      cookieString += `sameSite=${sameSite};`;
+    }
+
+    this.document.cookie = cookieString;
+  }
+
+
+  /**
+   * Verify if a cookie exists
+   *
+   * @param name - Cookie name
+   * @return boolean
+   */
+  public check(name: string): boolean {
+    if (!this.documentIsAccessible) {
+      return false;
+    }
+
+    const regExp: RegExp = this.getCookieRegExp(encodeURIComponent(name));
+    return regExp.test(this.document.cookie);
+  }
+
+
+  /**
+   * @param name Cookie name
+   * @returns any
+   */
+  public get(name: string): string {
+    if (this.documentIsAccessible && this.check(name)) {
+      const regExp: RegExp = this.getCookieRegExp(encodeURIComponent(name));
+      const result: RegExpExecArray | null = regExp.exec(this.document.cookie);
+
+      return result ? decodeURIComponent(result[1]) /* istanbul ignore next - Unreachable */ : '';
```

Comment:
if there is a result, could there be the case that its length is 1?

Reply:
I don't _think_ so..

I don't remember using `exec` before so I had to explore it a bit:

> If the match succeeds, the exec() method returns an array and updates properties of the regular expression object. The returned array has the matched text as the first item, and then one item for each capturing parenthesis that matched containing the text that was captured.

So it's pretty much `null` or 2+ items. I made a bin when I was trying to understand it: https://jsbin.com/pudaqan/edit?js,console